Sources/Orbit/Components/Dialog.swift

```diff
@@ -6,91 +6,109 @@ import SwiftUI
-        )
-        .overlay(
-            RoundedRectangle(cornerRadius: .small)
-                .strokeBorder(Self.shadowColor.opacity(0.15), lineWidth: BorderWidth.thin)
+                .shadow(color: Self.shadowColor.opacity(0.7), radius: .xxxLarge, x: 0, y: .xxxLarge / 2)
         )
+        .frame(maxWidth: Layout.readableMaxWidth / 2, maxHeight: .infinity)
+        .padding(.xLarge)
+        .background(Color.inkNormal.opacity(0.45))
+        .edgesIgnoringSafeArea(.all)
     }
-    
+
+    @ViewBuilder var buttons: some View {
+        switch buttonConfiguration {
+            case .primary(let label, let action):
+                Button(label, style: style.buttonStyle, action: action)
+            case .primaryAndSecondary(let primaryLabel, let secondaryLabel, let primaryAction, let secondaryAction):
+                Button(primaryLabel, style: style.buttonStyle, action: primaryAction)
+                ButtonLink(secondaryLabel, style: style.buttonLinkStyle, size: .button, action: secondaryAction)
+            case .primarySecondaryAndTertiary(let primaryLabel, let secondaryLabel, let tertiaryLabel, let primaryAction, let secondaryAction, let tertiaryAction):
+                Button(primaryLabel, style: style.buttonStyle, action: primaryAction)
+                ButtonLink(secondaryLabel, style: style.buttonLinkStyle, size: .button, action: secondaryAction)
+                ButtonLink(tertiaryLabel, style: style.buttonLinkStyle, size: .button, action: tertiaryAction)
+        }
+    }
+}
+
+// MARK: - Inits
+extension Dialog {
+
+    /// Creates Orbit Dialog component.
     public init(
         illustration: Illustration.Image = .none,
         title: String = "",
         description: String = "",
-        buttons: ButtonConfiguration
+        style: Style = .primary,
+        buttons: Buttons
     ) {
         self.illustration = illustration
         self.title = title
         self.description = description
-        self.buttons = buttons
+        self.style = style
+        self.buttonConfiguration = buttons
     }
 }
 
 // MARK: - Types
-public extension Dialog {
-    
-    enum ButtonConfiguration {
-        case primary(_ title: String, style: ButtonStyle = .normal, action: () -> Void = {})
+extension Dialog {
+
+    public enum Buttons {
+        case primary(_ title: String, action: () -> Void = {})
         case primaryAndSecondary(
-            primaryTitle: String,
-            secondaryTitle: String,
-            style: ButtonStyle = .normal,
+            primaryLabel: String,
+            secondaryLabel: String,
             primaryAction: () -> Void = {},
             secondaryAction: () -> Void = {}
         )
+        case primarySecondaryAndTertiary(
+            primaryLabel: String,
+            secondaryLabel: String,
+            tertiaryLabel: String,
+            primaryAction: () -> Void = {},
+            secondaryAction: () -> Void = {},
+            tertiaryAction: () -> Void = {}
```

Comment:
Maybe a `struct` could wrap each "button"? If we end up adding another property, it will be one more parameter per button... let's say we add accessibility identifiers... that would be 9 parameters in the this init.

```swift
struct Button {
    let title: String
    let action: () async -> Void = {}
    let accessibilityIdentifier: String
}
```
(but public, of course)

This would also group related properties instead of having them interleaved like it is now. I get the feeling that these parameters are optimized for the previews here in the package... in practice you would use all of them...

Reply:
`Button` might not be the best name though.

Reply:
Looking at it now, maybe we could dispense with `Buttons` and just have a variadic `init` :trollface:

Reply:
I read it after the update - moved that UI type to Button itself (possibly adding icon etc in future) to carry button UI type into these components that have buttons themselves. (to have `Button.Content` match `Icon.Content`)

Reply:
But will these buttons support icons? 🤔

Reply:
They can, it depends on the wrapper - not all wrappers will use the full future set of this Content, as most wrappers define/force the "look" of those buttons and mostly just need the "data" (icon/text/id/action).